lib/pacemaker/pcmk_sched_probes.c

```diff
@@ -446,6 +446,63 @@ add_probe_orderings_for_stops(pe_working_set_t *data_set)
     }
 }
 
+/*!
+ * \internal
+ * \brief Add necessary orderings between probe and starts of clone instances
+ *
+ * , in additon to the ordering with the parent resource added upon creating
+ * the probe.
+ *
+ * \param[in] probe     Probe as 'first' action in an ordering
+ * \param[in] after     'then' action wrapper in the ordering
+ */
+static void
+add_start_orderings_for_probe(pe_action_t *probe, pe_action_wrapper_t *after)
+{
+    uint32_t flags = pe_order_optional|pe_order_runnable_left;
+
+    if (after->action->rsc->variant <= pe_group
+        || uber_parent(probe->rsc) != after->action->rsc
+        || !pcmk__str_eq(after->action->task, RSC_START, pcmk__str_none)
+        || pcmk_is_set(probe->flags, pe_action_runnable)
+        // The order type is already enforced for its parent.
+        || pcmk_is_set(after->type, pe_order_runnable_left)) {
```

Comment:
for efficiency it's better to do the bit checks before the function calls

also a comment as to what these tests boil down to would be helpful

Reply:
Was thinking of readability. But yes, comments are helpful anyways :-)